mne/io/eeglab/eeglab.py

```diff
@@ -269,22 +315,42 @@ def __init__(self, input_fname, montage, preload=False, eog=(),
                 orig_format='double', verbose=verbose)
         else:
             if preload is False or isinstance(preload, string_types):
-                warn('Data will be preloaded. preload=False or a string '
-                     'preload is not supported when the data is stored in the '
-                     '.set file')
+                warnings.warn('Data will be preloaded. preload=False or a'
+                              ' string preload is not supported when the data'
+                              ' is stored in the .set file')
             # can't be done in standard way with preload=True because of
             # different reading path (.set file)
             data = eeg.data.reshape(eeg.nbchan, -1, order='F')
             data = data.astype(np.double)
             data *= CAL
+            data = np.vstack((data, self._event_ch))
             super(RawEEGLAB, self).__init__(
                 info, data, last_samps=last_samps, orig_format='double',
                 verbose=verbose)
 
+    def _add_trigger_ch(self, block, start, stop, sample_start, sample_stop):
+        """Callback function for adding the trigger channel."""
+        stim_ch = self._event_ch[start:stop][sample_start:sample_stop]
+        return np.vstack((block, stim_ch))
+
+    def _create_event_ch(self, events, n_sample=None):
+        """Create the event channel"""
+        if n_sample is None:
+            n_sample = self.last_samp - self.first_samp + 1
+        events = np.array(events, int)
+        if events.ndim != 2 or events.shape[1] != 3:
+            raise ValueError("[n_events x 3] shaped array required")
+        # update events
+        self._event_ch = _synthesize_stim_channel(events, n_sample)
+        self._events = events
+        if self.preload:
+            self._data[-1] = self._event_ch
+
```

Comment:
yeah a known issue. but I'm not sure if forcing consistency at this level is a priority. once you save them again you have a purified fif. In the case of doubt I would suggest to live with these inconsitencies. Each of these io APIs has a local culture. That's the pattern to expect.